documentation/usage/errorcodes/index.md

```diff
@@ -0,0 +1,74 @@
+
+
+## Validate Error Codes
+These error codes are surfaced when running `validate` or `validateWithResult`.
+
+### `SCHEMA_DOES_NOT_EXIST`
+- **Caused by:** The schema being validated against does not exist
+- **Solution:** Manually create the schema or enable [`createSchemas`](/documentation/configuration/parameters/createSchemas)
+
+### `FAILED_REPEATABLE_MIGRATION`
+- **Caused by:** A failed repeatable migration was detected
+- **Solution:** Remove any incompleted changes then run `repair` to fix the schema history
+
+### `FAILED_VERSIONED_MIGRATION`
+- **Caused by:** A failed versioned migration was detected
+- **Solution:** Remove any incompleted changes then run `repair` to fix the schema history
+
+### `APPLIED_REPEATABLE_MIGRATION_NOT_RESOLVED`
+- **Caused by:** A repeatable migration that was applied wasn't resolved in any supplied locations
+- **Solution:** If you removed this migration intentionally run `repair` to mark the migration as deleted
+
+### `APPLIED_VERSIONED_MIGRATION_NOT_RESOLVED`
+- **Caused by:** A versioned migration that was applied wasn't resolved in any supplied locations
+- **Solution:** If you removed this migration intentionally run `repair` to mark the migration as deleted
+
+### `RESOLVED_REPEATABLE_MIGRATION_NOT_APPLIED`
+- **Caused by:** A repeatable migration that was resolved has not been applied
+- **Solution:** To ignore this migration enable [`ignoreIgnoredMigrations`](/documentation/configuration/parameters/ignoreIgnoredMigrations)
+
+### `RESOLVED_VERSIONED_MIGRATION_NOT_APPLIED`
+- **Caused by:** A versioned migration that was resolved has not been applied
+- **Solution:** To ignore this migration enable [`ignoreIgnoredMigrations`](/documentation/configuration/parameters/ignoreIgnoredMigrations) and to allow executing this migration enable [`outOfOrder`](/documentation/configuration/parameters/outOfOrder)
+
+### `OUTDATED_REPEATABLE_MIGRATION`
+- **Caused by:** An applied repeatable migration was resolved with a newer checksum and can be reapplied
+- **Solution:** Run `migrate` to execute this migration
+
+### `TYPE_MISMATCH`
+- **Caused by:** The type of the resolved migration (`BASELINE`, `SQL`, `UNDO_SQL`, ...) is different from the applied migration's
+- **Solution:** Either revert the changes to the migration or run `repair` to update the schema history
+
+### `CHECKSUM_MISMATCH`
+- **Caused by:** The checksum of the resolved migration is different from the applied migration's
+- **Solution:** Either revert the changes to the migration or run `repair` to update the schema history
+
+### `DESCRIPTION_MISMATCH`
+- **Caused by:** The description of the resolved migration is different from the applied migration's
+- **Solution:** Either revert the changes to the migration or run `repair` to update the schema history
+
+
+## General Error Codes
+These error codes may appear from any command, and are indicative of more general errors.
+
+### `FAULT`
```

Comment:
I think this should be moved up to the top. `VALIDATE_ERROR` can live under the 'validate error codes' heading.